sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -1209,17 +1223,29 @@ __ESIMD_API std::enable_if_t<
         detail::is_device_accessor_with_v<AccessorT,
                                           detail::accessor_mode_cap::can_read>,
     simd<T, N>>
-block_load(AccessorT acc, PropertyListT props = {}) {
-  return block_load<T, N>(acc, 0, props);
+block_load(AccessorT acc, PropertyListT /* props */ = {}) {
+  // Create new properties without the alignment property passed in 'props',
+  // and add alignment<16> as it is usable and most favourable in this case.
+  constexpr auto L1Hint =
+      detail::getPropertyValue<PropertyListT, cache_hint_L1_key>(
+          cache_hint::none);
+  constexpr auto L2Hint =
+      detail::getPropertyValue<PropertyListT, cache_hint_L2_key>(
+          cache_hint::none);
+  static_assert(!PropertyListT::template has_property<cache_hint_L3_key>(),
+                "L3 cache hint is reserved. The old/experimental L3 LSC cache "
+                "hint is cache_level::L2 now.");
+  properties Props{cache_hint_L1<L1Hint>, cache_hint_L2<L2Hint>, alignment<16>};
```

Comment:
why do we want to ignore the properties-given alignment in this and the other cases?

Reply:
buffers/accessors should reference aligned memory on device. If byte_offset is zero we load from aligned device-buffer.
If user says the alignment is 4 it is unnecessarily pessimistic, which  will cause less efficient code-gen.
If alignment is more than 16, e.g. 256, it will not give any extra benefit to code-gen.

Reply:
silently ignoring it seems somewhat strange to me, could we static assert if it not 16 or something?

Reply:
having something like:
```c++
static_assert(!IsUserAlignedmentSpecified || UserAlignment >= 16, "Alignment is too pessimistic, specify 16 or more for more efficient code-gen");
```
seems too ... (annoying?).
Compiler reserves the right to optimize the code. This ignoring of smaller alignment is the optimization.

Reply:
My worry here is that we're silently ignoring explicit user specification that a user might expect to applied. We already silently ignore any other properties in the properties vector not related to ESIMD and just warn people in a comment but that seems reasonable to me since it's a totally different thing, if we have some ESIMD APIs that honor alignment and some that silently ignore it, that seems like it might be confusing to the user

Reply:
passing alignment<N> means "the address is aligned at least to "N"-bytes", but it may NOT be aligned at less than N-bytes (only N-bytes aligned guaranteed).

It is same as if you call USM aligned_alloc<T>(align=16, size); which means the requested memory must be aligned at at least 16-bytes, but the returned address may be 256-bytes aligned too (accidentally or intentionally).

In this case if user;s call of block_load<float, N>(acc, {alignement<4>}); means that user guarantees only 4-byte alignment. If by some heuristics we can guarantee better alignment, then why not use it?

Reply:
ok that convinces me, thanks